app/segments/drill/[id]/submission/input.js

```diff
@@ -281,6 +283,22 @@ function createOutputData(
   return outputData;
 }
 
+function invalidateOnSubmit(queryClient, did, currentTeamId, currentUserId) {
+  queryClient.invalidateQueries({
+    // used predicate as it seemed to be the best method to invalidate multiple query keys
+    predicate: (query) =>
+      query.queryKey[0] === "user" ||
```

Comment:
Is there a way (like object comparison) to do this or is this the only way? It's kinda unreadable...

Reply:
There's probably a cleaner way to do this, but the main thing I noticed is from here:

> `queryClient.invalidateQueries({ queryKey: ['key1', 'key2'] });` will invalidate all keys *starting with* `['key1', 'key2']`, not those starting with either `key1` or `key2`.

https://github.com/TanStack/query/issues/4669#issuecomment-1361210644

Another idea is to just write multiple invlidateQueries(), one for each key.

There are also some ideas here using stuff like Array.every:
https://www.reddit.com/r/reactjs/comments/17u2nll/tanstack_react_query_invalidatequeries/

Reply:
Marking this as resolved. Issue for maybe refactoring the invalidating multiple queries function into hook / Utility - https://github.com/Golf-Drill-Challenge-App/Golf-App/issues/149